lib/httparty/request.rb

```diff
@@ -276,11 +276,10 @@ def get_charset
     end
 
     def encode_with_ruby_encoding(body, charset)
-      if !body.nil? && Encoding.name_list.include?(charset)
-        body.force_encoding(charset)
-      else
-        body
-      end
+      encoding = Encoding.find(charset)
+      body.force_encoding(charset)
+    rescue ArgumentError
```

Comment:
Could you explain the change from checking body not being nil to rescuing ArgumentError? Thanks!